src/manager/node.c

```diff
@@ -1082,6 +1091,13 @@ static int agent_request_callback(sd_bus_message *m, void *userdata, UNUSED sd_b
         return req->cb(req, m, ret_error);
 }
 
+int agent_request_cancel(AgentRequest *r) {
+        _cleanup_agent_request_ AgentRequest *req = r;
+        req->is_cancelled = true;
+
+        return req->cb(req, NULL, NULL);
```

Comment:
I think you should pass an actual sd_bus_error (SD_BUS_ERROR_FAILED probably) here instead of NULL that says that the operation was cancelled.

Reply:
Ah, no the bus_error seems to be an out param, so you need to make a new error message with the error saying the operation was cancelled

Reply:
Done